Update remote local setup with hosts for end-to-end tests and IPv6 support

vicwicker opened the pull request:
**How to categorize this PR?**

/area dev-productivity
/kind enhancement

**What this PR does / why we need it**:

During startup, the remote local setup automatically updates `/etc/hosts` with entries needed for end-to-end tests. As more tests are added, new entries might be required, making the current `/etc/hosts` in the remote local setup outdated. This PR updates the setup to include the latest entries.

Additionally, it provides support for running the `prepare-operator-complete.sh` variant on IPv6. This variant runs the `operator-seed-up` setup. In short, `tmux` sessions can be created with `IPFAMILY=ipv6`, which passes the variable into the session and then to the KinD cluster when `make kind-multi-zone-up` and similar commands are executed.

**Special notes for your reviewer**:

/cc @istvanballok 

**Release note**:

```other developer
Update remote local setup with most recent hosts for end-to-end tests and instructions for an IPv6 setup
```

vicwicker committed:
Update /etc/hosts with new end-to-end test shoots

vicwicker committed:
Update /etc/hosts with entries for ci-e2e-kind-operator tests

vicwicker committed:
Add command to run IPv6 on "prepare-operator-complete.sh" setup

Only the "prepare-operator-complete.sh" setup supports IPv6 currently.

vicwicker committed:
Update /etc/hosts with IPv6 entries

vicwicker committed:
Deploy shoot-ipv6.yaml when IPv6 is used

vicwicker committed:
Use new "credentialsBindingName"

The "secretBindingName" field is deprecated in favor of
"credentialsBindingName".

vicwicker committed:
Support IPv6 for managed seed

This commit is analog to previous one in addition to introducing new
necessary files.

vicwicker committed:
Fix managed seed window panes

istvanballok commented on docs/deployment/content/remote-local-setup.yaml:
```diff
@@ -471,7 +493,7 @@ spec:
           new_window "shoot lifecycle" "Create" <<"EOF"
             export KUBECONFIG=$PWD/dev-setup/kubeconfigs/virtual-garden/kubeconfig
-            k apply -f example/provider-local/shoot.yaml
+            if [[ "$IPFAMILY" == "ipv6" ]]; then k apply -f example/provider-local/shoot-ipv6.yaml; else k apply -f example/provider-local/shoot.yaml; fi
```
Would it be cleaner to execute the `if` statement above the `new_window` function, such that the user experience, the tmux pane with the prepared commands matches exactly the desired scenario?

I verified that the following approach could work:

```bash
if  [[ "$IPFAMILY" == "ipv6" ]]; then
new_window "shoot lifecycle" "Create" <<"EOF"
  export KUBECONFIG=$PWD/dev-setup/kubeconfigs/virtual-garden/kubeconfig
  k apply -f example/provider-local/shoot-ipv6.yaml
EOF
else
new_window "shoot lifecycle" "Create" <<"EOF"
  export KUBECONFIG=$PWD/dev-setup/kubeconfigs/virtual-garden/kubeconfig
  k apply -f example/provider-local/shoot.yaml
EOF
fi
```

Similar comment could apply for the 2 other usages of `if` inside the `new_window` function.

istvanballok reviewed (commented):
/lgtm

I proposed a minor enhancement.

vicwicker committed:
Address review comments

This shifts IPv6 checks to occur during the creation of windows or
panes, and, thus, removing them from the window or pane command.

Co-authored-by: Istvan Zoltan Ballok <istvan.zoltan.ballok@sap.com>

istvanballok reviewed (commented):
/lgtm

ScheererJ reviewed (approved):
/assign

Thanks a lot for the nicely structured enhancement. Let's go IPv6 :-) .

/lgtm
/approve
